src/webgpu/api/validation/image_copy/buffer_related.spec.ts

```diff
@@ -0,0 +1,118 @@
+export const description = `Validation tests for buffer related parameters for copies`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import {
+  kSizedTextureFormats,
+  kTextureFormatInfo,
+  textureDimensionAndFormatCompatible,
+} from '../../../capability_info.js';
+import { kImageCopyTypes } from '../../../util/texture/layout.js';
+
+import { ImageCopyTest, formatCopyableWithMethod } from './image_copy.js';
+
+export const g = makeTestGroup(ImageCopyTest);
+
+g.test('valid')
+  .desc(`The buffer must be valid and not destroyed.`)
+  .params(u =>
+    u //
+      // B2B copy validations are at api,validation,encoding,cmds,copyBufferToBuffer.spec.ts
+      .combine('method', ['CopyB2T', 'CopyT2B'] as const)
+      .combine('bufferState', ['valid', 'destroyed', 'error'])
+  )
+  .fn(async t => {
+    const { method, bufferState } = t.params;
+
+    // A valid buffer.
+    let buffer = t.device.createBuffer({
+      size: 16,
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+    });
+
+    switch (bufferState) {
+      case 'destroyed': {
+        buffer.destroy();
+        break;
+      }
+      case 'error': {
+        buffer = t.getErrorBuffer();
+        break;
+      }
+    }
+
+    const success = bufferState === 'valid';
+    const submit = bufferState === 'destroyed';
```

Comment:
This might be simpler with `createEncoder` + `validateFinishAndSubmitGivenState`

Reply:
On second inspection I agree validateFinishAndSubmitGivenState won't work.